extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-sra.js

```diff
@@ -0,0 +1,471 @@
+import {
+  AmpA4A,
+  RENDERING_TYPE_HEADER,
+  XORIGIN_MODE,
+} from '../../../amp-a4a/0.1/amp-a4a';
+import {Services} from '../../../../src/services';
+import {
+  AmpAdNetworkDoubleclickImpl,
+  getNetworkId,
+  constructSRABlockParameters,
+  TFCD,
+  resetSraStateForTesting,
+} from '../amp-ad-network-doubleclick-impl';
+import {
+  MANUAL_EXPERIMENT_ID,
+} from '../../../../ads/google/a4a/traffic-experiments';
+import {
+  EXPERIMENT_ATTRIBUTE,
+} from '../../../../ads/google/a4a/utils';
+import {utf8Encode} from '../../../../src/utils/bytes';
+import {BaseElement} from '../../../../src/base-element';
+import {createElementWithAttributes} from '../../../../src/dom';
+import {layoutRectLtwh} from '../../../../src/layout-rect';
+import {Xhr, FetchResponseHeaders} from '../../../../src/service/xhr-impl';
+import {dev} from '../../../../src/log';
+import '../../../amp-ad/0.1/amp-ad';
+import * as sinon from 'sinon';
+
+function setupForAdTesting(win) {
+  const doc = win.document;
+  // TODO(a4a-cam@): This is necessary in the short term, until A4A is
+  // smarter about host document styling.  The issue is that it needs to
+  // inherit the AMP runtime style element in order for shadow DOM-enclosed
+  // elements to behave properly.  So we have to set up a minimal one here.
+  const ampStyle = doc.createElement('style');
+  ampStyle.setAttribute('amp-runtime', 'scratch-fortesting');
+  doc.head.appendChild(ampStyle);
+}
+
+describes.realWin('amp-ad-network-doubleclick-impl', {amp: true,
+  allowExternalResources: true}, env => {
+    let sandbox;
+    let doc;
+
+    beforeEach(() => {
+      setupForAdTesting(env.win);
+      doc = env.win.document;
+      sandbox = env.sandbox;
+      env.win.AMP_MODE.test = true;
+    });
+
+    describe('#SRA enabled', () => {
+      beforeEach(() => {
+        sandbox = env.sandbox;
+      });
+
+      it('should be disabled by default', () => {
+        const element = createElementWithAttributes(
+            doc, 'amp-ad', {
+              type: 'doubleclick',
+              height: 320,
+              width: 50,
+            });
+        doc.body.appendChild(element);
+        const impl = new AmpAdNetworkDoubleclickImpl(element);
+        expect(impl.useSra).to.be.false;
+      });
+
+      it('should force refresh off when enabled', () => {
+        const metaElement = createElementWithAttributes(doc, 'meta', {
+          name: 'amp-ad-doubleclick-sra',
+        });
+        doc.head.appendChild(metaElement);
+        const element = createElementWithAttributes(
+            doc, 'amp-ad', {
+              type: 'doubleclick',
+              height: 320,
+              width: 50,
+            });
+        doc.body.appendChild(element);
+        const impl = new AmpAdNetworkDoubleclickImpl(element);
+        expect(impl.useSra).to.be.true;
+        impl.layoutCallback();
+        expect(impl.refreshManager_).to.be.null;
+      });
+
+      it('should be enabled if meta tag present', () => {
```

Comment:
This is a subset of the previous, refresh disabled test.  Consider making one test?